src/Components/CommonComponent/MetaDataComponent.tsx

```diff
@@ -0,0 +1,44 @@
+import { useEffect } from 'react'
+import {
+  useGetCountriesMutation,
+  useGetRolesMutation,
+  useGetStatesMutation,
+} from '../../Services/MetaDataApi'
+import { MetaDataResponse } from '../../Type/ApiTypes'
+
+const useMetaData = () => {
+  const [getRoles] = useGetRolesMutation()
+  const [getCountries] = useGetCountriesMutation()
+  const [getStates] = useGetStatesMutation()
+
+  const fetchMetaData = async (getData: any) => {
+    try {
```

Comment:
for now onwards define these kind of API call functions in a separate file , 
increase re-usability

Reply:
yes , this is the separate file actually for implementing metadata apis